services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java

```diff
@@ -227,7 +406,40 @@ public final void setTenantInformationService(final TenantInformationService ten
                                 final RegistrationResult deviceResult = deviceInfoTracker.result();
                                 final RegistrationResult gatewayResult = gatewayInfoTracker.result();
 
-                                if (!isDeviceEnabled(deviceResult)) {
+                                if (deviceResult.isNotFound() && !gatewayResult.isNotFound()
+                                        && isDeviceEnabled(gatewayResult) && hasAuthorityForAutoRegistration(gatewayResult)) {
+
+                                    final Device device = new Device()
+                                        .setEnabled(true)
+                                        .setVia(Collections.singletonList(gatewayId));
+
+                                    final JsonArray memberOf = gatewayResult.getPayload().getJsonObject(RegistrationConstants.FIELD_DATA)
+                                            .getJsonArray(RegistryManagementConstants.FIELD_MEMBER_OF);
```

Comment:
Indeed. Since the code style is not enforced by the maven build I assumed it to be not "binding" especially since there is a lot of code which does not fully adhere to it.

I did the reformat but personally I did not find the result more readable than before 😒. Hence I would rather not apply it...

Reply:
The project code style guide aims to create a consistent code style irrespective of personal preferences which makes easier understanding of a large code base.  IMHO it is good to adhere to it.  As you said there are places where some code doesn't confirm to the code style and FMPOV those should also be updated whenever possible. @calohmn @sophokles73 WDYT?

Reply:
I think it would be good to adhere to the Hono code formatting style when possible, with the freedom to keep some lines with custom formatting. While I think some added linebreaks in the code in this PR for overly long lines wouldn't hurt, I don't see a direct need to change things.

Reply:
I applied the formatter in the latest commit. As I said, in my opinion it makes things worse... :-(

Reply:
I managed to find a middle ground between the formatter and readability together with @kaniyan.